v2/robotmk/src/bin/scheduler/internal_config.rs

```diff
@@ -88,14 +88,188 @@ pub fn sort_suites_by_id(suites: &mut [Suite]) {
+pub struct SystemEnvironment {}
+
+#[derive(Clone)]
+#[cfg_attr(test, derive(Debug, PartialEq))]
+pub struct RCCEnvironment {
+    pub binary_path: Utf8PathBuf,
+    pub robot_yaml_path: Utf8PathBuf,
+    pub controller: String,
+    pub space: String,
+    pub build_timeout: u64,
+    pub env_json_path: Option<Utf8PathBuf>,
+}
+
+impl Environment {
+    pub fn new(
+        suite_id: &str,
+        rcc_binary_path: &Utf8Path,
+        environment_config: &EnvironmentConfig,
+    ) -> Self {
+        match environment_config {
+            EnvironmentConfig::System => Self::System(SystemEnvironment {}),
+            EnvironmentConfig::Rcc(rcc_environment_config) => Self::Rcc(RCCEnvironment {
+                binary_path: rcc_binary_path.to_path_buf(),
+                robot_yaml_path: rcc_environment_config.robot_yaml_path.clone(),
+                controller: String::from("robotmk"),
+                space: suite_id.to_string(),
+                build_timeout: rcc_environment_config.build_timeout,
+                env_json_path: rcc_environment_config.env_json_path.clone(),
+            }),
+        }
+    }
+
+    pub fn wrap(&self, command_spec: CommandSpec) -> CommandSpec {
+        match self {
+            Self::System(system_environment) => system_environment.wrap(command_spec),
+            Self::Rcc(rcc_environment) => rcc_environment.wrap(command_spec),
+        }
+    }
+
+    pub fn create_result_code(&self, exit_code: i32) -> ResultCode {
+        match self {
+            Self::System(_) => SystemEnvironment::create_result_code(exit_code),
+            Self::Rcc(_) => RCCEnvironment::create_result_code(exit_code),
+        }
+    }
+}
+
+impl SystemEnvironment {
+    fn wrap(&self, command_spec: CommandSpec) -> CommandSpec {
+        command_spec
+    }
+
+    fn create_result_code(exit_code: i32) -> ResultCode {
+        if exit_code == 0 {
+            return ResultCode::AllTestsPassed;
+        }
+        ResultCode::RobotCommandFailed
+    }
+}
+
+impl RCCEnvironment {
+    fn wrap(&self, command_spec: CommandSpec) -> CommandSpec {
+        let mut wrapped_spec = CommandSpec::new(&self.binary_path);
+        wrapped_spec
+            .add_argument("task")
+            .add_argument("script")
+            .add_argument("--no-build");
+        apply_current_settings(
+            &self.robot_yaml_path,
+            &self.controller,
+            &self.space,
+            self.env_json_path.as_deref(),
+            &mut wrapped_spec,
+        );
+        wrapped_spec
+            .add_argument("--")
+            .add_argument(command_spec.executable)
+            .add_arguments(command_spec.arguments);
+        wrapped_spec
+    }
+
+    fn create_result_code(exit_code: i32) -> ResultCode {
+        match exit_code {
+            0 => ResultCode::AllTestsPassed,
+            10 => ResultCode::RobotCommandFailed,
+            _ => ResultCode::EnvironmentFailed,
+        }
+    }
+}
+
+pub fn apply_current_settings(
```

Comment:
It might be useful to make this a member function of `RCCEnvironment`, but let's to do this separately after merging

Reply:
Let's do this separately after merging

Reply:
https://github.com/elabit/robotmk/pull/456